openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -211,7 +211,8 @@ export class ResponseView {
             && !textFieldsIsNotBlank && !filesFiledIsNotBlank) {
         readyToSubmit = false;
       }
-      if (this.hasPendingUploadFiles() && !this.collectFilesDescriptions()) {
+      if (this.hasPendingUploadFiles()) {
+        this.collectFilesDescriptions();
```

Comment:
Preemptive note: we still need to call `collectFilesDescriptions()` each time because it's coupled with the `Upload` button enable logic. This just removes it's use for short-circuiting submit-ability.